src/core/impl/TChem_Impl_SingleParticleAerosolWater.hpp

```diff
@@ -0,0 +1,167 @@
+#ifndef __TCHEM_IMPL_SINGLEPARTICLEAEROSOLWATER_HPP__
+#define __TCHEM_IMPL_SINGLEPARTICLEAEROSOLWATER_HPP__
+
+#include "TChem_Util.hpp"
+
+namespace TChem {
+namespace Impl {
+
+  template<typename ValueType, typename DeviceType>
+struct AerosolWater_SingleParticle
+{
+    using value_type = ValueType;
+    using device_type = DeviceType;
+    using scalar_type = typename ats<value_type>::scalar_type;
+
+    using real_type = scalar_type;
+    /// sacado is value type
+    using value_type_1d_view_type = Tines::value_type_1d_view<value_type,device_type>;
+    using real_type_1d_view_type = Tines::value_type_1d_view<real_type,device_type>;
+    using ordinal_type_1d_view_type = Tines::value_type_1d_view<ordinal_type,device_type>;
+    using aerosol_model_data_type= AerosolModel_ConstData<device_type>;
+
+    using const_real_type_1d_view_type = Tines::value_type_1d_view<const real_type,device_type>;
+
+    template<typename MemberType>
+    KOKKOS_INLINE_FUNCTION static
+    void team_invoke(const MemberType& member,
+        ordinal_type i_part,
+        real_type number_conc,
+        real_type state[],  // TODO: figure out how the state vector actually works
+        //const AerosolModelData amd
+        const aerosol_model_data_type& amcd,
+        ordinal_type rh_idx,
+        ordinal_type aqueous_water_idx
+        //const aerosol_ion_pair_type ionpair,
+        //const std::map<std::string, int> ion_idx_map,
+        //const std::map<std::string, real_type> atomic_weights  // TODO: pack into ionpair objects
+        )
+    {   
+        // NOTE: Cannot use std::string with cuda 
+        // Declare and initialize variables for later use
+        
+        ordinal_type anion_idx;
+        ordinal_type cation_idx;
+        ordinal_type ion_species_idx;
+        ordinal_type qty;
+        std::string atom_name;
```

Comment:
We need to remove `std::string`. We will get a warning:

`#20014-D: calling a __host__ function from a __host__ __device__ function is not allowed`

when compiling the code in CUDA. Let's use an enum instead of `std::strin`g.